crates/driver/src/domain/competition/solution/mod.rs

```diff
@@ -322,3 +322,11 @@ pub enum SolutionError {
     #[error(transparent)]
     ProtocolFee(#[from] fee::Error),
 }
+
+#[derive(Debug, thiserror::Error)]
+pub enum MathError {
```

Comment:
nit: instead of calling all Errors `xyzError`, it would be nicer and inline with our goal to have short names with module qualifiers to have a module error and different types so we can refert to them as

error::Math, error::Solution, etc.

Reply:
I've refactored only the enums this PR touches.